cmd/frontend/internal/frontend/service.go

```diff
@@ -258,25 +258,30 @@ func (fes *FrontEndService) Monitor(ctx context.Context, errCh chan<- error) {
 		return
 	}
 
-	logForced := func() bool {
-		fes.monitorMu.Lock()
-		defer fes.monitorMu.Unlock()
-		forced := fes.logNextMonitorStatus
-		fes.logNextMonitorStatus = false
-		return forced
-	}
-
-	//linkCh := make(chan string, 1)
 	go func() {
-		var once sync.Once
-		//defer close(linkCh)
 		// status of external connectivity; requires 1 GW per IP family if configured to be reachable
-		init, noConnectivity := true, true
+		hasConnectivity := false
 		connectivityMap := map[string]bool{}
 		delay := 3 * time.Second // when started grant more time to write the whole config (minimize intial link flapping)
 		_ = fes.WaitStart(ctx)
 		var refreshCancel context.CancelFunc
+		denounce := true // Always denounce on container start
 		for {
+
+			if denounce {
+				denounce = false
+				hasConnectivity = false
+				if refreshCancel != nil {
+					refreshCancel()
+					refreshCancel = nil
+				}
+				if err := denounceFrontend(ctx, fes.targetRegistryClient); err != nil {
+					logger.Error(err, "denounceFrontend")
+				}
+				connectivityMap = map[string]bool{}
+				delay = 3 * time.Second // Avoid flapping (more?)
```

Comment:
Not sure about this one. An env var would be nice.

Reply:
ok

Reply:
Honestly I have no idea what those errors I called "session errors" are and what that can cause them. So, please take a close look, and suggest what to do differently if needed